src/neo4j_graphrag/experimental/pipeline/kg_builder.py

```diff
@@ -0,0 +1,236 @@
+                on_error=self.on_error,
+                prompt_template=self.prompt_template,
+            ),
+            "extractor",
+        )
+        pipe.add_component(self.kg_writer, "writer")
+
+        if self.from_pdf:
+            pipe.add_component(self.pdf_loader, "pdf_loader")
+
+            pipe.connect(
+                "pdf_loader",
+                "splitter",
+                input_config={"text": "pdf_loader.text"},
+            )
+
+            pipe.connect(
+                "schema",
+                "extractor",
+                input_config={
+                    "schema": "schema",
+                    "document_info": "pdf_loader.document_info",
+                },
+            )
+        else:
+            pipe.connect(
+                "schema",
+                "extractor",
+                input_config={
+                    "schema": "schema",
+                },
+            )
+
+        pipe.connect(
+            "splitter",
+            "extractor",
+            input_config={"chunks": "splitter"},
+        )
+
+        # Connect extractor to writer
+        pipe.connect(
+            "extractor",
+            "writer",
+            input_config={"graph": "extractor"},
+        )
+
+        return pipe
+
+    async def run_async(
+        self, file_path: Optional[str] = None, text: Optional[str] = None
+    ) -> PipelineResult:
+        """
+        Asynchronously runs the knowledge graph building process.
+
+        Args:
+            file_path (Optional[str]): The path to the PDF file to process. Required if `from_pdf` is True.
+            text (Optional[str]): The text content to process. Required if `from_pdf` is False.
+
+        Returns:
+            PipelineResult: The result of the pipeline execution.
+        """
+        pipe_inputs = self._prepare_inputs(file_path=file_path, text=text)
+        return await self.pipeline.run(pipe_inputs)
+
+    def run(
+        self, file_path: Optional[str] = None, text: Optional[str] = None
+    ) -> PipelineResult:
+        """
+        Runs the knowledge graph building process.
+
+        Args:
+            file_path (Optional[str]): The path to the PDF file to process. Required if `from_pdf` is True.
+            text (Optional[str]): The text content to process. Required if `from_pdf` is False.
+
+        Returns:
+            PipelineResult: The result of the pipeline execution.
+        """
+        return asyncio.run(self.run_async(file_path=file_path, text=text))
+
+    def _prepare_inputs(
+        self, file_path: Optional[str], text: Optional[str]
+    ) -> dict[str, Any]:
+        if self.from_pdf:
+            if file_path is None or text is not None:
+                raise ValueError(
+                    "Expected 'file_path' argument when 'from_pdf' is True."
+                )
+        else:
+            if text is None or file_path is not None:
+                raise ValueError("Expected 'text' argument when 'from_pdf' is False.")
```

Comment:
Maybe we can reuse `PipelineDefinitionError` here?

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/165/commits/24c6f0aa60007040e4b0a55a6e4a273518df9f52